tests/unit/composition/PieChart.test.ts

```
@@ -0,0 +1,148 @@
+import type { Composition } from '$lib'
+import { describe, expect, test, vi } from 'vitest'
+
+// Mock the composition parsing utilities
+vi.mock(`$lib/composition/parse`, () => ({
+  composition_to_percentages: vi.fn((comp: Composition) => {
+    const total = Object.values(comp).reduce((sum, val) => sum + (val || 0), 0)
+    const percentages: Composition = {}
+    for (const [element, amount] of Object.entries(comp)) {
+      if (typeof amount === `number`) {
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)